cassiopeia/data/CassiopeiaTree.py

```diff
@@ -2018,3 +2018,38 @@ def get_unmutated_characters_along_edge(
     def copy(self) -> "CassiopeiaTree":
         """Full copy of CassiopeiaTree"""
         return copy.deepcopy(self)
+
+    def impute_unambiguous_missing_states(self):
+        """
+        Impute unambiguous missing states.
+
+        If a state is missing in a node but present in its parent,
+        then it can be imputed unambiguously as the parent's state.
+        We perform all these imputations.
+
+        Raises:
+            CassiopeiaTreeError if the character vectors do not all have
+            the same length, or if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+        for (parent, child) in self.depth_first_traverse_edges():
+            parent_states = self.get_character_states(parent)
+            child_states = self.get_character_states(child)
+            if not len(parent_states) == len(child_states):
+                raise CassiopeiaTreeError(
+                    "Parent and child node have different length character "
+                    "states."
+                )
```

Comment:
I haven't thought about branch length estimation for conflicting states. This comment also applies to the recently merged #124  . Maybe we can create a new PR to raise an error if there are conflicting states? This does bring up the point that not all methods can handle all kinds of data, and as the CassiopeiaTree grows to support more and more kinds of data (like conflicting states), older methods that don't know about this will blow up. Maintenance can become a bit of a nightmare. Don't know what the best way deal with this is.

Reply:
Won't do anything about this ATM